pkg/tsdb/postgres/postgres.go

```diff
@@ -69,28 +96,22 @@ func generateConnectionString(datasource *models.DataSource, logger log.Logger)
 		// Attach root certificate if provided
 		if sslRootCert := datasource.JsonData.Get("sslRootCertFile").MustString(""); sslRootCert != "" {
 			logger.Debug("Setting server root certificate", "sslRootCert", sslRootCert)
-			sslOpts = fmt.Sprintf("%s&sslrootcert=%s", sslOpts, url.QueryEscape(sslRootCert))
+			connStr += fmt.Sprintf(" sslrootcert='%s'", sslRootCert)
 		}
 
 		// Attach client certificate and key if both are provided
 		sslCert := datasource.JsonData.Get("sslCertFile").MustString("")
 		sslKey := datasource.JsonData.Get("sslKeyFile").MustString("")
 		if sslCert != "" && sslKey != "" {
 			logger.Debug("Setting SSL client auth", "sslCert", sslCert, "sslKey", sslKey)
-			sslOpts = fmt.Sprintf("%s&sslcert=%s&sslkey=%s", sslOpts, url.QueryEscape(sslCert), url.QueryEscape(sslKey))
+			connStr += fmt.Sprintf(" sslcert='%s' sslkey='%s'", sslCert, sslKey)
 		} else if sslCert != "" || sslKey != "" {
 			return "", fmt.Errorf("SSL client certificate and key must both be specified")
 		}
 	}
 
-	u := &url.URL{
-		Scheme: "postgres",
-		User:   url.UserPassword(datasource.User, datasource.DecryptedPassword()),
-		Host:   datasource.Url, Path: datasource.Database,
-		RawQuery: sslOpts,
-	}
-
-	return u.String(), nil
+	logger.Debug("Generated Postgres connection string successfully")
```

Comment:
I don't find this very useful. We already have https://github.com/grafana/grafana/blob/615b000acd0ae301a159e24bdc6e32f870aaf974/pkg/tsdb/postgres/postgres.go#L32-L34